CHANGELOG.md

```diff
@@ -3,6 +3,723 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 8.0.0-alpha.0
+
+<!-- generated comparing v7.22.1..master -->
+
+_Nov 7, 2024_
+
+We'd like to offer a big thanks to the 21 contributors who made this release possible. Here are some highlights ✨:
+
+TODO INSERT HIGHLIGHTS
+
+From https://github.com/mui/mui-x/pull/14913
+
+### Tree View
+
+- The `ContentComponent` or `ContentProps` props of the `<TreeItem />` component have been removed in favor of the new `slots`, `slotProps` props and of the `useTreeItem` hook.
+
+  Learn more about the anatomy of the Tree Items and the customization utilities provided on the [Tree Item Customization page](https://next.mui.com/x/react-tree-view/tree-item-customization/).
+
+- The `onClick` and `onMouseDown` were the only event callback that were passed to the content of the Tree Item instead of its root.
+  The goal was to make sure that the callback was not fired when clicking on a descendant of a giving item.
+  This inconsistency has been solved, all the event manager now target the root of the item, and you can use the `onItemClick` prop on the Tree View component to target the content of an item:
+
+  ```diff
+  -<SimpleTreeView>
+  +<SimpleTreeView onItemClick={handleItemClick}>
+  -  <TreeItem onClick={handleItemClick}>
+  +  <TreeItem >
+   </SimpleTreeView>
+  ```
+
+- All the new Tree Item-related components and utils (introduced in the previous major to improve the DX of the Tree Item component) are becoming the default way of using the Tree Item and are therefore losing their `2` suffix:
+
+  ```diff
+   import * as React from 'react';
+   import {
+  -  TreeItem2,
+  +  TreeItem,
+  -  TreeItem2Root,
```

Comment:
@flaviendelangle is all this information necessary? It feels like we are not going straight to the point and that these should be kept in the PR or migration guide.

Reply:
So far AFAIK we always added the exact same content on the migration guide and on the changelog/release note.

If we want to stop doing it and instead add a summary + a link to the migration guide any time it's a longer section, I'm fine doing it.

WDYT @cherniavskii @alexfauquette @LukasTy @joserodolfofreitas?

Reply:
I guess we could keep the "introduction text" as is and replace all the examples with a link to the migration guide.

Reply:
I guess it's only for the `alpha.0` that the BCs take a bit more than expected space, from next versions, since there'll be a release every week, it shouldn't be as long.

Reply:
i've removed the examples, let me know if the wording is wrong or where we should keep the examples.

 I tried to proofread where I removed them, but you know better than I what was necessary or not